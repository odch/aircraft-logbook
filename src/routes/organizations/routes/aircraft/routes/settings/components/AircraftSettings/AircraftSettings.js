import React from 'react'
import PropTypes from 'prop-types'
import { Redirect } from 'react-router-dom'
import withStyles from '@material-ui/core/styles/withStyles'
import Typography from '@material-ui/core/Typography'
import isLoaded from '../../../../../../../../util/isLoaded'
import {
  organization as organizationShape,
  aircraft as aircraftShape
} from '../../../../../../../../shapes'
import LoadingIcon from '../../../../../../../../components/LoadingIcon'

const styles = theme => ({
  container: {
    padding: '1em',
    [theme.breakpoints.up(1000 + theme.spacing.unit * 3 * 2)]: {
      width: 1000,
      marginLeft: 'auto',
      marginRight: 'auto'
    }
  }
})

class AircraftSettings extends React.Component {
  componentDidMount() {
    const { organization, fetchAircrafts } = this.props

    if (organization) {
      fetchAircrafts(organization.id)
    }
  }

  componentDidUpdate(prevProps) {
    const { organization, fetchAircrafts } = this.props

    if (
      organization &&
      (!prevProps.organization || prevProps.organization.id !== organization.id)
    ) {
      fetchAircrafts(organization.id)
    }
  }

  render() {
    const { organization, aircraft, classes } = this.props

    if (!isLoaded(organization) || !isLoaded(aircraft)) {
      return <LoadingIcon />
    }

    if (organization === null) {
      return <Redirect to="/" />
    }

    if (aircraft === null) {
      return <Redirect to={`/organizations/${organization.id}`} />
    }

    return (
      <div className={classes.container}>
        <Typography variant="display1" gutterBottom>
          {aircraft.registration}
        </Typography>
      </div>
    )
  }
}

AircraftSettings.propTypes = {
  organization: organizationShape,
  aircraft: aircraftShape,
  classes: PropTypes.object.isRequired,
  fetchAircrafts: PropTypes.func.isRequired
}

export default withStyles(styles)(AircraftSettings)
