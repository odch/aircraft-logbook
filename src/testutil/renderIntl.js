import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
