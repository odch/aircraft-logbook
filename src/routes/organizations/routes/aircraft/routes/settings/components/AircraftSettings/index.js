import AircraftSettings from './AircraftSettings'

export default AircraftSettings
