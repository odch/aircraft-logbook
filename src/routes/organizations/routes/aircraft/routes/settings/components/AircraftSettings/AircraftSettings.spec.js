import React from 'react'
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import renderIntl from '../../../../../../../../testutil/renderIntl'
import AircraftSettings from './AircraftSettings'

const StartPage = () => <div>Start Page</div>
const OrganizationPage = () => <div>Organization Page</div>

describe('routes', () => {
  describe('organizations', () => {
    describe('routes', () => {
      describe('aircraft', () => {
        describe('routes', () => {
          describe('settings', () => {
            describe('components', () => {
              describe('AircraftSettings', () => {
                it('renders loading icon if organization not loaded', () => {
                  const tree = renderIntl(
                    <AircraftSettings
                      organization={undefined}
                      fetchAircrafts={() => {}}
                    />
                  ).toJSON()
                  expect(tree).toMatchSnapshot()
                })

                it('renders loading icon if aircraft not loaded', () => {
                  const tree = renderIntl(
                    <AircraftSettings
                      organization={{ id: 'my_org' }}
                      aircraft={undefined}
                      fetchAircrafts={() => {}}
                    />
                  ).toJSON()
                  expect(tree).toMatchSnapshot()
                })

                it('redirects to start page if organization was not found', () => {
                  const tree = renderIntl(
                    <Router>
                      <Switch>
                        <Route exact path="/" component={StartPage} />
                        <AircraftSettings
                          organization={null}
                          fetchAircrafts={() => {}}
                        />
                      </Switch>
                    </Router>
                  ).toJSON()
                  expect(tree).toMatchSnapshot()
                })

                it('redirects to organization page if aircraft was not found', () => {
                  const tree = renderIntl(
                    <Router>
                      <Switch>
                        <Route
                          exact
                          path="/organizations/my_org"
                          component={OrganizationPage}
                        />
                        <AircraftSettings
                          organization={{ id: 'my_org' }}
                          aircraft={null}
                          fetchAircrafts={() => {}}
                        />
                      </Switch>
                    </Router>
                  ).toJSON()
                  expect(tree).toMatchSnapshot()
                })

                it('renders aircraft settings when everything is loaded', () => {
                  const store = configureStore()({
                    firestore: {
                      data: {
                        organizationAircrafts: {
                          o7flC7jw8jmkOfWo8oyA: {
                            settings: {
                              fuelTypes: [
                                { name: 'avgas', description: 'AvGas' },
                                { name: 'mogas', description: 'MoGas' }
                              ]
                            }
                          }
                        }
                      }
                    },
                    aircraftSettings: {
                      createCheckDialog: {
                        open: false
                      },
                      deleteCheckDialog: {
                        open: false
                      },
                      createFuelTypeDialog: {
                        open: false
                      }
                    }
                  })
                  const tree = renderIntl(
                    <Provider store={store}>
                      <Router>
                        <AircraftSettings
                          organization={{ id: 'my_org' }}
                          aircraft={{
                            id: 'o7flC7jw8jmkOfWo8oyA',
                            registration: 'HBKFW'
                          }}
                          fetchAircrafts={() => {}}
                        />
                      </Router>
                    </Provider>
                  ).toJSON()
                  expect(tree).toMatchSnapshot()
                })

                it('calls fetchAircrafts when mounted with organization', () => {
                  const fetchAircrafts = jest.fn()

                  renderIntl(
                    <AircraftSettings
                      organization={{ id: 'my_org' }}
                      aircraft={undefined}
                      fetchAircrafts={fetchAircrafts}
                    />
                  )

                  expect(fetchAircrafts).toBeCalledWith('my_org')
                })

                it('does not call fetchAircrafts when mounted without organization', () => {
                  const fetchAircrafts = jest.fn()

                  renderIntl(
                    <AircraftSettings
                      organization={undefined}
                      aircraft={undefined}
                      fetchAircrafts={fetchAircrafts}
                    />
                  )

                  expect(fetchAircrafts).not.toBeCalled()
                })

                it('calls fetchAircrafts when updated with new organization', () => {
                  const fetchAircrafts = jest.fn()

                  const renderer = renderIntl(
                    <AircraftSettings
                      organization={undefined}
                      aircraft={undefined}
                      fetchAircrafts={fetchAircrafts}
                    />
                  )

                  expect(fetchAircrafts).not.toBeCalled()

                  renderer.update(
                    <AircraftSettings
                      organization={{ id: 'my_org' }}
                      aircraft={undefined}
                      fetchAircrafts={fetchAircrafts}
                    />
                  )

                  expect(fetchAircrafts).toBeCalledWith('my_org')
                })
              })
            })
          })
        })
      })
    })
  })
})
