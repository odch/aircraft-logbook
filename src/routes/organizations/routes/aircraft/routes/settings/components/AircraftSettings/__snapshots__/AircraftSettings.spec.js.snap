// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes organizations routes aircraft routes settings components AircraftSettings redirects to aircraft page if not manager or techlog manager 1`] = `
<div>
  Aircraft Page
</div>
`;

exports[`routes organizations routes aircraft routes settings components AircraftSettings redirects to organization page if aircraft was not found 1`] = `
<div>
  Organization Page
</div>
`;

exports[`routes organizations routes aircraft routes settings components AircraftSettings redirects to start page if organization was not found 1`] = `
<div>
  Start Page
</div>
`;

exports[`routes organizations routes aircraft routes settings components AircraftSettings renders aircraft settings when everything is loaded 1`] = `
<div
  className="injectIntl(AircraftSettings)-container-1"
>
  <h4
    className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
  >
    HBKFW
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root injectIntl(AircraftSettings)-closeSettingsButton-4 MuiButton-text MuiButton-textPrimary"
      href="/organizations/my_org/aircrafts/o7flC7jw8jmkOfWo8oyA"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      <span
        className="MuiButton-label"
      >
        Einstellungen schliessen
      </span>
    </a>
  </h4>
  <div>
    <h5
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      Checks
    </h5>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Check hinzuf체gen
      </span>
    </button>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
    >
      Keine Checks erfasst
    </p>
  </div>
  <div>
    <h5
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      Treibstofftypen
    </h5>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Treibstofftyp hinzuf체gen
      </span>
    </button>
    <ul
      className="MuiList-root MuiList-dense MuiList-padding"
    >
      <li
        className="MuiListItem-container"
      >
        <div
          className="MuiListItem-root MuiListItem-dense MuiListItem-secondaryAction"
          disabled={false}
        >
          <div
            className="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
            >
              AvGas
            </span>
            <p
              className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              avgas
            </p>
          </div>
        </div>
        <div
          className="MuiListItemSecondaryAction-root"
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
          </button>
        </div>
      </li>
      <li
        className="MuiListItem-container"
      >
        <div
          className="MuiListItem-root MuiListItem-dense MuiListItem-secondaryAction"
          disabled={false}
        >
          <div
            className="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body2"
            >
              MoGas
            </span>
            <p
              className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              mogas
            </p>
          </div>
        </div>
        <div
          className="MuiListItemSecondaryAction-root"
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
          </button>
        </div>
      </li>
    </ul>
  </div>
  <h5
    className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
  >
    Erweiterte Einstellungen
  </h5>
  <ul
    className="MuiList-root MuiList-padding"
  >
    <li
      className="MuiListItem-container"
    >
      <div
        className="MuiListItem-root MuiListItem-secondaryAction"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Flugstunden-Z채hler aktiviert
          </span>
        </div>
      </div>
      <div
        className="MuiListItemSecondaryAction-root"
      >
        <span
          className="MuiSwitch-root MuiSwitch-edgeEnd"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-123 MuiSwitch-switchBase MuiSwitch-colorSecondary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <span
                className="MuiSwitch-thumb"
              />
              <input
                checked={false}
                className="PrivateSwitchBase-input-126 MuiSwitch-input"
                onChange={[Function]}
                type="checkbox"
              />
            </span>
          </span>
          <span
            className="MuiSwitch-track"
          />
        </span>
      </div>
    </li>
    <li
      className="MuiListItem-container"
    >
      <div
        className="MuiListItem-root MuiListItem-secondaryAction"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Motorstunden-Z채hler aktiviert
          </span>
        </div>
      </div>
      <div
        className="MuiListItemSecondaryAction-root"
      >
        <span
          className="MuiSwitch-root MuiSwitch-edgeEnd"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-123 MuiSwitch-switchBase MuiSwitch-colorSecondary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <span
                className="MuiSwitch-thumb"
              />
              <input
                checked={false}
                className="PrivateSwitchBase-input-126 MuiSwitch-input"
                onChange={[Function]}
                type="checkbox"
              />
            </span>
          </span>
          <span
            className="MuiSwitch-track"
          />
        </span>
      </div>
    </li>
    <li
      className="MuiListItem-container"
    >
      <div
        className="MuiListItem-root MuiListItem-secondaryAction"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            Techlog aktiviert
          </span>
        </div>
      </div>
      <div
        className="MuiListItemSecondaryAction-root"
      >
        <span
          className="MuiSwitch-root MuiSwitch-edgeEnd"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-123 MuiSwitch-switchBase MuiSwitch-colorSecondary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <span
                className="MuiSwitch-thumb"
              />
              <input
                checked={false}
                className="PrivateSwitchBase-input-126 MuiSwitch-input"
                onChange={[Function]}
                type="checkbox"
              />
            </span>
          </span>
          <span
            className="MuiSwitch-track"
          />
        </span>
      </div>
    </li>
  </ul>
  <hr
    className="MuiDivider-root injectIntl(AircraftSettings)-divider-2"
  />
  <div
    className="injectIntl(AircraftSettings)-deleteButtonContainer-3"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text"
      data-cy="aircraft-delete-button"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
        Flugzeug l철schen
      </span>
    </button>
  </div>
</div>
`;

exports[`routes organizations routes aircraft routes settings components AircraftSettings renders loading icon if aircraft not loaded 1`] = `
<div
  className="LoadingIcon-container-5"
>
  <div
    className="MuiCircularProgress-root LoadingIcon-icon-6 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;

exports[`routes organizations routes aircraft routes settings components AircraftSettings renders loading icon if organization not loaded 1`] = `
<div
  className="LoadingIcon-container-5"
>
  <div
    className="MuiCircularProgress-root LoadingIcon-icon-6 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;

exports[`routes organizations routes aircraft routes settings components AircraftSettings renders only certain fields for techlogmanager 1`] = `
<div
  className="injectIntl(AircraftSettings)-container-1"
>
  <h4
    className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
  >
    HBKFW
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root injectIntl(AircraftSettings)-closeSettingsButton-4 MuiButton-text MuiButton-textPrimary"
      href="/organizations/my_org/aircrafts/o7flC7jw8jmkOfWo8oyA"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      <span
        className="MuiButton-label"
      >
        Einstellungen schliessen
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </a>
  </h4>
  <div>
    <h5
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      Checks
    </h5>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Check hinzuf체gen
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
    >
      Keine Checks erfasst
    </p>
  </div>
</div>
`;
